src/repr/src/bytes.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use mz_ore::cast::CastLossy;
+
+/// Struct for postgres compatible size units which is slightly different from
+/// [`bytesize::ByteSize`]. Instead of MiB or GiB and so on, it uses MB, GB for the sizes
+/// with 1024 multiplier.
+/// Valid units are B, kB, MB, GB, TB, PB with multiples of 1024
+/// where 1MB = 1024kB
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize, Default)]
```

Comment:
deriving `proptest_derive::Arbitrary` here would be great! Then we can very easily get some randomized testing coverage